python/tank/bootstrap/resolver.py

```diff
@@ -194,9 +204,75 @@ def resolve_configuration(self, config_descriptor, sg_connection):
                 self._bundle_cache_fallback_paths
             )
 
+    def find_matching_pipelines(self, pipeline_config_name, current_login, sg_connection):
+        """
+        Finds all the pipelines matching the query parameters.
+
+        :param str pipeline_config_name: Name of the pipeline configuration requested for. If ``None``,
+            all pipeline configurations from the project will be matched.
+        :param str current_login: Only retains non-primary configs from the specified user.
+        :param ``shotgun_api3.Shotgun`` sg_connection: Connection to the Shotgun site.
+
+        :returns iterator: Returns an iterator over the pipeline configurations matching the
```

Comment:
FWIW I think we should always ask ourselves "do I really need to return a collection or can I  simply yield the results". The former makes the code more bloated by having to declare a collection we need to add to and then return, while yielding provides shorter code. It's not about performance but about code flow.